clients/drcachesim/tests/memref_gen.h

```diff
@@ -207,8 +207,10 @@ add_encodings_to_memrefs(instrlist_t *ilist,
             const int instr_size = instr_length(GLOBAL_DCONTEXT, pair.instr);
             pair.memref.instr.addr = offset + base_addr;
             pair.memref.instr.size = instr_size;
-            memcpy(pair.memref.instr.encoding, &decode_buf[offset], instr_size);
-            pair.memref.instr.encoding_is_new = true;
+            if (!set_only_instr_addr) {
+                memcpy(pair.memref.instr.encoding, &decode_buf[offset], instr_size);
+                pair.memref.instr.encoding_is_new = true;
```

Comment:
Is your code assuming encoding_is_new is always initialized?  E.g., "if (!use_module_mapper_ && memref_instr.encoding_is_new) {" in decode_cache.h?

Reply:
In that condition, encoding_is_new will be read only if the user didn't ask to use the module mapper, in which case the init() time check will ensure the trace filetype supports embedded-encodings, in which case encoding_is_new will be set.